src/coreclr/jit/compiler.hpp

```diff
@@ -2688,6 +2688,12 @@ inline var_types Compiler::mangleVarArgsType(var_types type)
             default:
                 break;
         }
+
+        if (varTypeIsSIMD(type))
+        {
+            // Vectors also get passed in int registers. Use TYP_INT.
+            return TYP_INT;
```

Comment:
Is the size mismatch here fine?

Reply:
It turns out the size doesn't matter; the only thing that matters is INT (use general-purpose registers) or non-INT (use floaing-point/SIMD registers).